src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -336,6 +340,14 @@ def update_managed_disk(cmd, instance, size_gb=None, sku=None, disk_iops_read_wr
         instance.disk_iops_read_write = disk_iops_read_write
     if disk_mbps_read_write is not None:
         instance.disk_mbps_read_write = disk_mbps_read_write
+    if disk_encryption_set is not None:
+        if not is_valid_resource_id(disk_encryption_set):
+            disk_encryption_set = resource_id(
+                subscription=get_subscription_id(cmd.cli_ctx), resource_group=resource_group_name,
+                namespace='Microsoft.Compute', type='diskEncryptionSets', name=disk_encryption_set)
+        instance.encryption.disk_encryption_set_id = disk_encryption_set
+    if encryption_type is not None:
```

Comment:
is `encrytion_type` dependent on `disk_encryption_set`? that means  if `disk_encrytpion_set` is None, but `encrytion_type` is not None, update will fail

Reply:
I add a check now.